erpnext/stock/doctype/stock_entry/stock_entry.py

```diff
@@ -100,7 +100,7 @@ def validate_item(self):
 					if f in ["stock_uom", "conversion_factor"] or not item.get(f):
 						item.set(f, item_details.get(f))
 
-			if self.difference_account:
+			if self.difference_account and item.expense_account=="":
```

Comment:
Thanks for the pull request. Can you change the condition like the following:

`if self.difference_account and not item.expense_account:`

This will work for both null and blank value.